vpn/src/main/java/com/duckduckgo/mobile/android/vpn/breakage/ReportBreakageCategorySingleChoiceActivity.kt

```diff
@@ -0,0 +1,170 @@
+                HtmlCompat.FROM_HTML_MODE_LEGACY
+            )
+    }
+
+    private fun configureListeners() {
+        val categories = viewModel.shuffledCategories.map { getString(it.category) }.toTypedArray()
+        binding.categoriesSelection.setOnClickListener {
+            AlertDialog.Builder(this)
+                .setTitle(getString(R.string.atp_ReportBreakageCategoriesTitle))
+                .setSingleChoiceItems(categories, viewModel.indexSelected) { _, newIndex ->
+                    viewModel.onCategoryIndexChanged(newIndex)
+                }
+                .setPositiveButton(getString(android.R.string.yes)) { dialog, _ ->
+                    viewModel.onCategoryAccepted()
+                    dialog.dismiss()
+                }
+                .setNegativeButton(getString(android.R.string.no)) { dialog, _ ->
+                    viewModel.onCategorySelectionCancelled()
+                    dialog.dismiss()
+                }
+                .show()
+        }
+        binding.ctaNextFormSubmit.setOnClickListener { viewModel.onSubmitPressed() }
+    }
+
+    private fun configureObservers() {
+        lifecycleScope.launch {
+            viewModel.commands()
+                .flowWithLifecycle(lifecycle, STARTED)
+                .collectLatest { processCommand(it) }
+        }
+        lifecycleScope.launch {
+            viewModel.viewState()
+                .flowWithLifecycle(lifecycle, STARTED)
+                .collectLatest { render(it) }
+        }
+    }
+
+    private fun processCommand(command: Command) {
+        when (command) {
+            Command.ConfirmAndFinish -> confirmAndFinish()
+        }
+    }
+
+    private fun confirmAndFinish() {
+        lifecycleScope.launch {
+            val issue =
+                IssueReport(
+                    appName = brokenApp.appName,
+                    appPackageId = brokenApp.appPackageId,
+                    description = binding.appBreakageFormFeedbackInput.text.toString(),
+                    category = viewModel.viewState.value.categorySelected.toString(),
+                    customMetadata =
+                    Base64.encodeToString(
+                        metadataReporter.getVpnStateMetadata(brokenApp.appPackageId).toByteArray(),
+                        Base64.NO_WRAP or Base64.NO_PADDING or Base64.URL_SAFE
+                    )
+                )
+            deviceShieldPixels.sendAppBreakageReport(issue.toMap())
+            setResult(RESULT_OK, Intent().apply { issue.addToIntent(this) })
+            finish()
+        }
+    }
+
+    private fun render(viewState: ViewState) {
+        val category =
+            viewState.categorySelected?.let { getString(viewState.categorySelected.category) }.orEmpty()
+        binding.categoriesSelection.setText(category)
+        if (viewState.indexSelected == 8) binding.otherCategoryDescription.visibility = View.VISIBLE else View.GONE
```

Comment:
Small change here as the visibility doesn't change to GONE after is made VISIBLE:

`binding.otherCategoryDescription.visibility = if (viewState.indexSelected == 8) View.VISIBLE else View.GONE`

Reply:
Whoops! How sloppy of me 😬 -- fixed now, thanks!